src/PostgREST/Middleware.hs

```diff
@@ -69,74 +63,50 @@ runPgLocals conf claims app req = do
       setConfigLocal mempty ("search_path", schemas)
     preReqSql = (\f -> "select " <> toS f <> "();") <$> configDbPreRequest conf
 
--- | Log in apache format. Only requests that have a status greater than minStatus are logged.
--- | There's no way to filter logs in the apache format on wai-extra: https://hackage.haskell.org/package/wai-extra-3.0.29.2/docs/Network-Wai-Middleware-RequestLogger.html#t:OutputFormat.
--- | So here we copy wai-logger apacheLogStr function: https://github.com/kazu-yamamoto/logger/blob/a4f51b909a099c51af7a3f75cf16e19a06f9e257/wai-logger/Network/Wai/Logger/Apache.hs#L45
--- | TODO: Add the ability to filter apache logs on wai-extra and remove this function.
-pgrstFormat :: Status -> OutputFormatter
-pgrstFormat minStatus date req status responseSize =
-  if status < minStatus
-    then mempty
-  else toLogStr (getSourceFromSocket req)
-    <> " - - ["
-    <> toLogStr date
-    <> "] \""
-    <> toLogStr (requestMethod req)
-    <> " "
-    <> toLogStr (rawPathInfo req <> rawQueryString req)
-    <> " "
-    <> toLogStr (show (httpVersion req)::Text)
-    <> "\" "
-    <> toLogStr (show (statusCode status)::Text)
-    <> " "
-    <> toLogStr (maybe "-" show responseSize::Text)
-    <> " \""
-    <> toLogStr (fromMaybe mempty $ requestHeaderReferer req)
-    <> "\" \""
-    <> toLogStr (fromMaybe mempty $ requestHeaderUserAgent req)
-    <> "\"\n"
-  where
-    getSourceFromSocket = BS.pack . showSockAddr . remoteHost
+    unquoted :: JSON.Value -> Text
+    unquoted (JSON.String t) = t
+    unquoted (JSON.Number n) =
+      toS $ formatScientific Fixed (if isInteger n then Just 0 else Nothing) n
+    unquoted (JSON.Bool b) = show b
+    unquoted v = toS $ JSON.encode v
 
-pgrstMiddleware :: LogLevel -> Application -> Application
+pgrstMiddleware :: LogLevel -> Middleware
 pgrstMiddleware logLevel =
-    logger
+    logger logLevel
   . gzip def
   . cors corsPolicy
-  . staticPolicy (only [("favicon.ico", "static/favicon.ico")])
-  where
-    logger = case logLevel of
-      LogCrit  -> id
-      LogError -> unsafePerformIO $ mkRequestLogger def { outputFormat = CustomOutputFormat $ pgrstFormat status500}
-      LogWarn  -> unsafePerformIO $ mkRequestLogger def { outputFormat = CustomOutputFormat $ pgrstFormat status400}
-      LogInfo  -> logStdout
 
-defaultCorsPolicy :: CorsResourcePolicy
-defaultCorsPolicy =  CorsResourcePolicy Nothing
-  ["GET", "POST", "PATCH", "PUT", "DELETE", "OPTIONS"] ["Authorization"] Nothing
-  (Just $ 60*60*24) False False True
+logger :: LogLevel -> Middleware
+logger logLevel app req sendResponse = app req $ \res ->
+  let
+    status = responseStatus res
+    logRequest = logStdout app req sendResponse
+  in
+  case logLevel of
+    LogInfo                        -> logRequest
+    LogWarn  | status >= status400 -> logRequest
+    LogError | status >= status500 -> logRequest
+    _                              -> sendResponse res
```

Comment:
Nitpick, more explicit:

```suggestion
    LogCrit                        -> sendResponse res
```

Reply:
Hm. Just replacing `_` with `LogCrit` won't work, because `_` also catches the failing conditions for `LogWarn` and `LogError`. I think the `_` catchall fits best.

Reply:
True. Didn't saw that one. It's good to leave it as it is then.